runtime/resolvers/legacy_metrics.go

```diff
@@ -75,12 +82,20 @@ func newLegacyMetrics(ctx context.Context, opts *runtime.ResolverOptions) (runti
 		return nil, fmt.Errorf("failed to build query: %w", err)
 	}
 
+	// Extract fields and row filter from the query using the queries.SecurityFromQuery helper
+	rowFilter, fields, err := queries.SecurityFromQuery(props.QueryName, props.QueryArgsJSON)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract accessible fields: %w", err)
+	}
```

Comment:
Consider moving to `InferRequiredSecurityRules` since it's only used there (then it doesn't need new struct vars). Would require support for returning an `error` from `InferRequiredSecurityRules` – would that be a problem?